tests/integration/targets/mso_schema_site_contract_service_graph/tasks/main.yml

```diff
@@ -0,0 +1,770 @@
+
+- name: Query all site contract service graphs without passing the service_graph and contract name
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    state: query
+  ignore_errors: true
+  register: query_all_sg_without_contract_sg
+
+- name: Assertions check for the query results
+  assert:
+    that:
+      - query_sg_with_wrong_name is not changed
+      - query_sg_with_wrong_name.current == {}
+      - query_empty_contract_without_sg_name is not changed
+      - query_empty_contract_without_sg_name.current == {}
+      - query_invalid_contract_sg is not changed
+      - query_invalid_contract_sg.current == {}
+      - query_invalid_contract_sg.msg == "The site contract{{':'}} Contract4 does not exist."
+      - query_all_sg_with_empty_template is not changed
+      - query_all_sg_with_empty_template.current == {}
+      - query_sg_without_name is not changed # Need to rerun the below task
+      - query_sg_without_name.current != {}
+      - query_sg_without_name.current.serviceGraphRef is match(".+/templates/ansible_template1/serviceGraphs/ansible_service_graph")
+      - query_all_sg_without_contract_sg is not changed
+      - query_all_sg_without_contract_sg.current.0.serviceGraphRef != ""
+
+- name: Remove the site contract Contract1 - service graph from the ansible_template1 - check_mode
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract1
+    state: absent
+  check_mode: true
+  ignore_errors: true
+  register: site_con_sg_absent_cm
+
+- name: Remove the site contract Contract1 - service graph from the ansible_template1 - normal_mode
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract1
+    state: absent
+  ignore_errors: true
+  register: site_con_sg_absent_nm
+
+- name: Remove the site contract Contract1 - service graph from the ansible_template1 - normal_mode again
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract1
+    state: absent
+  ignore_errors: true
+  register: site_con_sg_absent_nm_again
+
+- name: Assertions check for remove the site contract Contract1 - service graph from the ansible_template1
+  assert:
+    that:
+      - site_con_sg_absent_cm is changed
+      - site_con_sg_absent_cm.current == {}
+      - site_con_sg_absent_cm.previous != {}
+      - site_con_sg_absent_nm is changed
+      - site_con_sg_absent_nm.current == {}
+      - site_con_sg_absent_nm.previous != {}
+      - site_con_sg_absent_nm_again is not changed
+      - site_con_sg_absent_nm_again.current == {}
+      - site_con_sg_absent_nm_again.previous == {}
```

Comment:
Are we not checking any attributes anymore on delete? Also missing proposed. Applies to all asserts for delete